semgrep-core/src/tainting/Dataflow_tainting.ml

```diff
@@ -383,128 +412,171 @@ let handle_taint_propagators env x taints =
         Taints.union taints_in_acc taints_strid)
       Taints.empty propagate_tos
   in
-  let taints = Taints.union taints taints_incoming in
-  let var_env =
+  let lval_env =
     match x with
-    | `Var var ->
-        (* If `x` is a variable, then taint is propagated by side-effect. This
-         * allows us to e.g. propagate taint from `x` to `y` in `f(x,y)`. *)
-        Lval_env.add_var var taints_incoming var_env
+    (* If `x` is a variable, then taint is propagated by side-effect. This
+     * allows us to e.g. propagate taint from `x` to `y` in `f(x,y)`. *)
+    | `Lval lval when LV.lval_is_var_and_dots lval ->
```

Comment:
I take note that this is unclear and I'll add a TODO for a refactoring, maybe should add a task too in Linear. This may involve encapsulating more of this logic inside `Taint_lval_env`. I'm adding some more comments for now. But to clarify this, we are only field sensitive on lvalues of the form `x.a.b.c` which is what `lval_is_var_and_dots` is testing.

In this function we are processing `pattern-propagators`, so anything could be propagated, I was just trying to provide an example of taint-propagation "by side effect". I will have to work on taint doc for a while, bear with me.